src/evict/evict_lru.c

```diff
@@ -699,8 +699,11 @@ __evict_pass(WT_SESSION_IMPL *session)
          */
         if (!WT_EVICT_HAS_WORKERS(session) &&
           (cache->evict_empty_score < WT_EVICT_SCORE_CUTOFF ||
-            !__evict_queue_empty(cache->evict_urgent_queue, false)))
+            !__evict_queue_empty(cache->evict_urgent_queue, false))) {
+            F_CLR(session, WT_SESSION_LOCKED_PASS);
             WT_RET(__evict_lru_pages(session, true));
```

Comment:
In case of error, we never set the `WT_SESSION_LOCKED_PASS`. Capture the return value in a local variable and set the flag and verify the return value.